### PR TITLE
add --keep-env flag 

### LINKED ISSUE #12485: ENH(remove): Add option to keep the (empty) environment despite removal of all packages
### What is the idea?

Conda environments are more than a list of packages. Metadata like channel priority, subdir etc. are also a crucial part.

However, the CLI seems to assume that empty environments serve no purpose - this is implicit in the behaviour of `conda remove`: when you remove all packages, it deletes the environment: `conda remove --all -n ENV_NAME` not just removes all packages, it _also_ removes the environment and its metadata.

Also, when you remove all packages without explicitly passing `--all` the environment itself _with_ the metadata is removed, despite this not necessarily being what the user wants. This has been discussed before in https://github.com/conda/conda/issues/10467

I would like to suggest a CLI option for `conda remove` that tells `conda remove` _not_ to delete the environment even if all packages are uninstalled.

This is consistent with environment creation: `conda create` can create _empty_ environments, so the existence of empty environments _is_ valid. Hence there's no reason why one shouldn't go back to empty environments from a full environment with remove.

### What should happen?

Ideally, the behaviour of `conda remove` would be changed to not delete the environment if all packages are gone. But since this is quite a deep change, it may be easier to add a new option that changes that default behaviour: `conda remove --all --keep-env -n ENV_NAME` could be how one tells remove to not remove the environment.

This should also work for `conda remove --keep-env scipy` in cases where remove would remove all packages without the shortcut `--all` being passed.

### Additional Context

See this StackOverflow question for further context: https://stackoverflow.com/questions/75731916/how-to-remove-all-packages-from-a-conda-environment-but-not-keep-environment-i

### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
### Description

Resolves #12485 

Adding a new `--keep-env` flag to be used with `conda remove --all`. It allows users to remove all the packages in the environment while retaining the environment itself.

- [x] Add a `--keep-env` flag to retain the environment when using `conda remove --all`
- [x] Create a new module `test_main_remove.py`. 
- [x] Move `test_remove_all` from `test/test_create.py` to this new module.
- [x] Use `tmp_env()` instead of `make_temp_env()` and `conda_cli()` instead of `run_command()`.
- [x] delete the original test from `tests/test_create.py`
- [x] Write test for `keep-env` flag.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: tests/cli/test_main_remove.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+)
+from conda.gateways.disk.delete import path_is_clean
+from conda.testing import CondaCLIFixture, TmpEnvFixture
+from conda.testing.integration import (
+    PYTHON_BINARY,
+    package_is_installed,
+)
+
+
+def test_remove_all(tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture):
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:21]: This is moved from `tests/test_create.py`, remember to delete the original

File: tests/cli/test_main_remove.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    with tmp_env("python") as prefix:
+        assert exists(join(prefix, PYTHON_BINARY))
+        assert package_is_installed(prefix, "python")
+
+        # regression test for #2154
+        with pytest.raises(PackagesNotFoundError) as exc:
+            conda_cli("remove", f"--prefix={prefix}", "python", "foo", "numpy")
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:28]: `conda_cli` does not magically insert `--yes` like the older `run_command` implementation did so we need to be explicit

```suggestion
            conda_cli("remove", f"--prefix={prefix}", "python", "foo", "numpy", "--yes")
```

File: tests/cli/test_main_remove.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        exception_string = repr(exc.value)
+        assert "PackagesNotFoundError" in exception_string
+        assert "- numpy" in exception_string
+        assert "- foo" in exception_string
+
+        conda_cli("remove", f"--prefix={prefix}", "--all")
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:34]:
```suggestion
        conda_cli("remove", f"--prefix={prefix}", "--all", "--yes")
```

File: tests/cli/test_main_remove.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        assert path_is_clean(prefix)
+
+
+def test_remove_all_keep_env(tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture):
+    with tmp_env("python") as prefix:
+        assert exists(join(prefix, PYTHON_BINARY))
+        assert package_is_installed(prefix, "python")
+
+        conda_cli("remove", f"--prefix={prefix}", "--all", "--keep-env")
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:43]:
```suggestion
        conda_cli("remove", f"--prefix={prefix}", "--all", "--keep-env", "--yes")
```

File: tests/cli/test_main_remove.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from os.path import (
+    exists,
+    join,
+)
+
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:7]:
```suggestion
```

File: tests/cli/test_main_remove.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import pytest
+
+from conda.exceptions import (
+    PackagesNotFoundError,
+)
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:12]:
```suggestion
from conda.exceptions import PackagesNotFoundError
```

File: tests/cli/test_main_remove.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+from conda.gateways.disk.delete import path_is_clean
+from conda.testing import CondaCLIFixture, TmpEnvFixture
+from conda.testing.integration import (
+    PYTHON_BINARY,
+    package_is_installed,
+)
```
Comment (kenodegard) [inline tests/cli/test_main_remove.py:18]:
```suggestion
from conda.testing.integration import PYTHON_BINARY, package_is_installed
```

**Review by kenodegard — CHANGES_REQUESTED** [review]
`tmp_env` returns a `pathlib.Path` object so we should use it as such

**Comment by travishathaway** [comment]
@ForgottenProgramme,

I think it would be really nice to add an example to the help page:

https://github.com/conda/conda/blob/349ca85ce327ed76cfa514498f36d050c946b1ec/conda/cli/main_remove.py#L43-L58

**Review by jezdez — APPROVED** [review]

**Review by travishathaway — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Review by kenodegard — APPROVED** [review]

**Pull request merged**